### ISSUE TITLE
[BUG] Cpp2 contracts need Cpp1 _condition_ semantics

### ISSUE THREAD

**Issue by JohelEGP** [hsutter/cppfront#377]
An `explicit operator bool()` is not a candidate as currently implemented.

**Minimal reproducer** (https://godbolt.org/z/PGbz4b47T):
```Cpp2
struct t {
    explicit operator bool();
};
f: () [[pre: t()]] = { }
main: () = { }
```

**Commands**:
```bash
cppfront x.cpp2
clang++17 -std=c++2b -stdlib=libc++ -I $CPPFRONT_INCLUDE_DIR x.cpp
```

**Expected result**: A well-formed program by ensuring the condition is contextually converted to `bool`.

**Actual result and error**:

<details><summary>Generated C++1.</summary>

```C++
//=== Cpp2 type declarations ====================================================


#include "cpp2util.h"


//=== Cpp2 type definitions and function declarations ===========================

struct t {
    explicit operator bool();
};
#line 4 "/app/main.cpp2"
auto f() -> void;
auto main() -> int;

//=== Cpp2 function definitions =================================================


#line 4 "/app/main.cpp2"
auto f() -> void{
                       cpp2::Default.expects(t(), "");
#line 4 "/app/main.cpp2"
}
auto main() -> int{}
```

</details>

```output
main.cpp2:5:46: error: no viable conversion from 't' to 'bool'
                       cpp2::Default.expects(t(), "");
                                             ^~~
build/_cppfront/main.cpp:12:14: note: explicit conversion function is not a candidate
    explicit operator bool();
             ^
cpp2util.h:325:37: note: passing argument to parameter 'b' here
    constexpr auto expects    (bool b, CPP2_MESSAGE_PARAM msg = "" CPP2_SOURCE_LOCATION_PARAM_WITH_DEFAULT)
                                    ^
1 error generated.
```

**See also**: <https://quuxplusone.github.io/blog/2023/04/08/most-ctors-should-be-explicit/#explicit-operator-bool-const>.

**Comment by JohelEGP** [comment]
Not actually part of <https://wg21.link/P2521>.

**Comment by JohelEGP** [comment]
Implemented by commit 2893e64cae356f01721deede8d488dc7e47a3b4c: <https://cpp2.godbolt.org/z/7nnYc8oG9>.
